source/node.h

```diff
@@ -0,0 +1,106 @@
+#include <memory>
+#include <iterator>
+#include <unordered_map>
+#include <any>
+#include <vector>
+
+namespace uast {
+    class Node {
```

Comment:
We can change the style, but it would be har, cause all the IDLE's use such indentation by default, but they can be customized, of course. Should we do it anyway?